src/storage_volume.h

```diff
@@ -74,10 +74,10 @@ class StorageVolume : public NLVObject<virStorageVolPtr, StorageVolumeCleanupHan
   NLV_PRIMITIVE_RETURN_WORKER(Delete, virStorageVolPtr, bool);
 
   // ACCESSOR/MUTATOR WORKERS
-  NLV_PRIMITIVE_RETURN_WORKER(GetKey, virStorageVolPtr, std::string);
-  NLV_PRIMITIVE_RETURN_WORKER(GetName, virStorageVolPtr, std::string);
-  NLV_PRIMITIVE_RETURN_WORKER(GetPath, virStorageVolPtr, std::string);
-  NLV_PRIMITIVE_RETURN_WORKER(ToXml, virStorageVolPtr, std::string);
+  NLV_STRING_RETURN_WORKER(GetKey, virStorageVolPtr, std::string);
```

Comment:
nitpick: I would say leave it as PRIMITIVE if you are going to keep accepting the type as parameter, unless I'm misunderstanding something...